lib/executor/ramping_arrival_rate.go

```diff
@@ -300,21 +301,20 @@ func (varr RampingArrivalRate) Run(ctx context.Context, out chan<- stats.SampleC
 	}).Debug("Starting executor run...")
 
 	activeVUsWg := &sync.WaitGroup{}
-	defer activeVUsWg.Wait()
 
+	returnedVUs := make(chan struct{})
 	startTime, maxDurationCtx, regDurationCtx, cancel := getDurationContexts(ctx, duration, gracefulStop)
-	defer cancel()
-
 	// Pre-allocate the VUs local shared buffer
-	activeVUs := make(chan lib.ActiveVU, maxVUs)
-	activeVUsCount := uint64(0)
-	// Make sure all VUs aren't executing iterations anymore, for the cancel()
-	// above to deactivate them.
+
 	defer func() {
-		for i := uint64(0); i < activeVUsCount; i++ {
-			<-activeVUs
-		}
+		// Make sure all VUs aren't executing iterations anymore, for the cancel()
+		// above to deactivate them.
```

Comment:
```suggestion
		// below to deactivate them.
```

But I'd suggest removing this comment altogether since the flow is clear now and no longer depends on the `defer` order.